demos/ios/MASVS-RESILIENCE/MASTG-DEMO-0023/MastgTest.swift

```diff
@@ -0,0 +1,113 @@
+import SwiftUI
+import MachO
+
+struct MastgTest {
+    static func checkDebuggerEntitlement() -> Bool {
+        print("\n=== Starting Debug Check ===\n")
+        
+        // Method 1: Check Bundle Info
+        print("Method 1: Checking Bundle Info")
+        let bundleURL = Bundle.main.bundleURL
+        print("📦 Bundle URL: \(bundleURL.path)")
+        
+        // Try to read Info.plist
+        let infoPath = bundleURL.appendingPathComponent("Info.plist").path
+        print("📄 Info.plist Path: \(infoPath)")
+        
+        if let infoDict = NSDictionary(contentsOfFile: infoPath) {
+            print("📋 Info.plist Contents: \(infoDict)")
+        }
+        
+        // Method 2: Check Executable
+        print("\nMethod 2: Checking Executable")
+        if let executablePath = Bundle.main.executablePath {
+            print("📍 Executable Path: \(executablePath)")
+            
+            do {
+                let fileManager = FileManager.default
+                let attributes = try fileManager.attributesOfItem(atPath: executablePath)
+                print("📊 File Attributes: \(attributes)")
+                
+                // Try reading with different methods
+                let data = try Data(contentsOf: URL(fileURLWithPath: executablePath))
+                print("📄 File Size: \(data.count) bytes")
+                
+                if data.count > 0 {
+                    if let content = String(data: data, encoding: .ascii),
+                       content.contains("get-task-allow") {
```

Comment:
I am afraid this method might cause false positives. For example, if there is a third-party code in the binary that also contains `get-task-allow`, this would trigger a false positive. It's also a bit slow to search through the whole binary, no? Can you optimise it?